src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java

```diff
@@ -322,18 +322,8 @@ public void postBuildStatus(@NonNull BitbucketBuildStatus status) throws IOExcep
      */
     @Override
     public boolean checkPathExists(@NonNull String branchOrHash, @NonNull String path) throws IOException {
-        StringBuilder encodedPath = new StringBuilder(path.length() + 10);
-        boolean first = true;
-        for (String segment : StringUtils.split(path, "/")) {
-            if (first) {
-                first = false;
-            } else {
-                encodedPath.append('/');
-            }
-            encodedPath.append(Util.rawEncode(segment));
-        }
-        int status = getRequestStatus(String.format(API_BROWSE_PATH, getUserCentricOwner(), repositoryName, encodedPath,
-                URLEncoder.encode(branchOrHash, "UTF-8")));
+        int status = getRequestStatus(String.format(API_BROWSE_PATH, getUserCentricOwner(), repositoryName,
```

Comment:
This should have been duplicate code removal, but instead it revealed the bug in `encodePath` which was appending the '/' instead of prepending it